osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuHitObject.cs

```diff
@@ -14,6 +16,28 @@ public class DrawableOsuHitObject : DrawableHitObject<OsuHitObject>
         public const float TIME_FADEIN = 400;
         public const float TIME_FADEOUT = 500;
 
+        public double FadeInSpeedMultiplier = 1;
+        public double FadeOutSpeedMultiplier = 1;
+
+        private double preemptFadeOut;
+
+        /// <summary>
+        /// The number of milliseconds before <see cref="HitObject.StartTime"/> that we should fade out. Clamped between 0 and 1000 milliseconds.
+        /// </summary>
+        public double PreemptFadeOut
+        {
+            get => preemptFadeOut;
+            set
+            {
+                if (preemptFadeOut == value)
+                    return;
+
+                preemptFadeOut = MathHelper.Clamp(value, 0, 1000);
+            }
+        }
+
+        public override bool IsPresent => base.IsPresent || State.Value == ArmedState.Idle && Time.Current >= HitObject.StartTime - TIME_PREEMPT;

```

Comment:
This is pretty weird, why is this needed?

Reply:
to stop invisible circles from not receiving input